File: app/src/main/java/com/suhang/networkmvp/ui/activity/SplashActivity.java
package com.suhang.networkmvp.ui.activity;

import android.os.Bundle;

import com.suhang.networkmvp.R;
import com.suhang.networkmvp.annotation.ActivityScope;
import com.suhang.networkmvp.binding.data.BaseData;
import com.suhang.networkmvp.dagger.module.BlankModule;
import com.suhang.networkmvp.databinding.ActivitySplashBinding;
import com.suhang.networkmvp.mvp.model.BlankModel;

@ActivityScope
public class SplashActivity extends BaseActivity<BlankModel,ActivitySplashBinding>{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected int bindLayout() {
        return R.layout.activity_splash;
    }

    @Override
    protected void subscribeEvent() {

    }

    @Override
    protected void initData() {
    }

    @Override
    protected BaseData getBindingData() {
        return null;
    }

    @Override
    protected void injectDagger() {
        getBaseComponent().getBlankComponent(new BlankModule()).inject(this);
    }
}

File: app/src/main/java/com/suhang/networkmvp/ui/activity/SplashActivity.kt
package com.suhang.networkmvp.ui.activity

import android.os.Bundle
import com.suhang.layoutfinderannotation.GenSubComponent
import com.suhang.networkmvp.R
import com.suhang.networkmvp.application.DaggerHelper
import com.suhang.networkmvp.constants.Constant
import com.suhang.networkmvp.dagger.module.BlankModule
import com.suhang.networkmvp.mvp.model.BlankModel

class SplashActivity : BaseActivity<BlankModel>() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bind(R.layout.activity_splash)
    }


    override fun subscribeEvent() {

    }

    override fun initData() {}

}
